lib/ansible/modules/cloud/online/online_server_install.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: online_server_install
+short_description: Install an Online server
+description:
+  - Gather facts about the servers.
+  - U(https://www.online.net/en/dedicated-server)
+version_added: "2.8"
+author:
+  - "Ludovic Petetin (@broferek)"
+extends_documentation_fragment: online
+'''
+
+EXAMPLES = r'''
+- name: Install Centos on the server 5824 with the user foo created
+  online_server_install:
+    api_token: '0d1627e8-bbf0-44c5-a46f-5c4d3aef033f'
+    host: "5824"
+    os_id: 305
+    hostname: "sd-5824"
+    root_password: "root_bar"
+    user_login: "foo"
+    user_password: "bar"
+    partitioning_template_ref: "a645ex65-af43-4rr8-bc45-654qsdf654e2"
+'''
+
+RETURN = r'''
+---
+msg:
+    description: Message specifying that the installation has started
+    returned: success
+    type: string
+    sample: The installation has started. Check your server state at https://api.online.net/fr/server/state/5824
```

Comment:
Ansible module are supposed to give a JSON back. Could you add more information id, and other relevant attributes you would get in the payload of the request?

Reply:
The API doesn't return anything except a code result, hence, if the call is successful, the ouput message will redirect the user to the console so that he can follow the installation of his server.
Could you be more precise about what you would like to have in the output?

Reply:
I'm still not sure to understand what you mean here.

I use the normal module.exit_json function in case of a module success.
    module.exit_json(
        changed=True,
        msg="The installation has started. Check your server state at {0}/fr/server/state/{1}".format(module.params.get('api_url'), host)
    )
So, it outputs JSON as you would expect.

The documentation only says that inside the JSON returned, there is a field called "msg" which is a string but the output itself is not a string.

I can add another 'status' field in the exit JSON if you want. Maybe something like this would be better ?
    module.exit_json(
        changed=True,
        status='installing',
        msg="The installation has started. Check your server state at {0}/fr/server/state/{1}".format(module.params.get('api_url'), host)
    )

I can't say "status installed" as you said as the API returns a 200 code when the installation starts, it doesn't wait or the server to be actually installed. I guess the user will have to wait for the SSH port to be opened in another task to consider its server installed.

Reply:
I'm still not sure to understand what you mean here.

I use the normal module.exit_json function in case of a module success.
    module.exit_json(
        changed=True,
        msg="The installation has started. Check your server state at {0}/fr/server/state/{1}".format(module.params.get('api_url'), host)
    )
So, it outputs JSON as you would expect.

The documentation only says that inside the JSON returned, there is a field called "msg" which is a string but the output itself is not a string.

I can add another 'status' field in the exit JSON if you want. Maybe something like this would be better ?
    module.exit_json(
        changed=True,
        status='installing',
        msg="The installation has started. Check your server state at {0}/fr/server/state/{1}".format(module.params.get('api_url'), host)
    )

I can't say "status installed" as you said as the API returns a 200 code when the installation starts, it doesn't wait or the server to be actually installed. I guess the user will have to wait for the SSH port to be opened in another task to consider its server installed.

Reply:
I'm still not sure to understand what you mean here.

I use the normal module.exit_json function in case of a module success.
```python
    module.exit_json(
        changed=True,
        msg="The installation has started. Check your server state at {0}/fr/server/state/{1}".format(module.params.get('api_url'), host)
    )
```
So, it outputs JSON as you would expect.

The documentation only says that inside the JSON returned, there is a field called "msg" which is a string but the output itself is not a string.

I can add another 'status' field in the exit JSON if you want. Maybe something like this would be better ?
```python
    module.exit_json(
        changed=True,
        status='installing',
        msg="The installation has started. Check your server state at {0}/fr/server/state/{1}".format(module.params.get('api_url'), host)
    )
```

I can't say "status installed" as you said as the API returns a 200 code when the installation starts, it doesn't wait or the server to be actually installed. I guess the user will have to wait for the SSH port to be opened in another task to consider its server installed.